packages/server/lib/util/replace_stream.js

```diff
@@ -0,0 +1,40 @@
+const GraphemeSplitter = require('grapheme-splitter')
+const through = require('through')
+
+const splitter = new GraphemeSplitter()
+
+// UTF-8 grapheme aware stream replacer
+// https://github.com/cypress-io/cypress/pull/4984
+function replaceStream (pattern, replacement, options = { maxTailLength: 100 }) {
+  let tail = ''
+
+  return through(function write (chunk) {
+    chunk = chunk.toString('utf8')
+
+    tail = tail + chunk
+    tail = tail.replace(pattern, replacement)
```

Comment:
Without knowing much about Node.js streams or what is actually replaced here, I wonder how are replacements handled that fall over the edge of a chunk?

Reply:
I'm not talking about splitting Unicode graphemes on the chunk boundaries, I'm talking about the string replacement here. If the matching string has the first half in one chunk and the second half in the next, the `tail.replace(pattern, replacement)` won't find it. 'chunk boundary friendliness' is one of the features of the original _replacestream_ library.

Reply:
After staring at the code some more I see what you mean. Sorry to have bothered you.